### PR TITLE
Shift swap request Slack follow-ups

### PR DESCRIPTION
**Pull request by vadimkerr**
# What this PR does

Add Slack follow-up messages for shift swap requests:

Screenshot 2023-08-15 at 20 19 49

## Which issue(s) this PR fixes

https://github.com/grafana/oncall/issues/2679

## Checklist

- [x] Unit, integration, and e2e (if applicable) tests updated
- [x] Documentation added (or `pr:no public docs` PR label added if not required)
- [x] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: engine/settings/prod_without_db.py [hunk 1]
```diff
@@ -77,8 +77,10 @@ def on_uwsgi_worker_exit():
     "apps.schedules.tasks.notify_about_empty_shifts_in_schedule.start_notify_about_empty_shifts_in_schedule": {
         "queue": "default"
     },
-    "apps.schedules.tasks.shift_swaps.slack_messages.post_shift_swap_request_creation_message": {"queue": "default"},
```
Comment (joeyorlando) [inline engine/settings/prod_without_db.py:80]: good catch 👍

File: engine/apps/schedules/tasks/shift_swaps/slack_followups.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        if not _has_followup_been_sent(shift_swap_request):
+            send_shift_swap_request_slack_followup.delay(shift_swap_request.pk)
+            _mark_followup_sent(shift_swap_request)
+
+
+def _get_shift_swap_requests_in_followup_window(now: datetime.datetime) -> list[ShiftSwapRequest]:
+    """Get all SSRs that are in the followup window."""
+
+    shift_swap_requests_in_notification_window = []
+    for shift_swap_request in ShiftSwapRequest.objects.filter(benefactor__isnull=True, swap_start__gt=now):
```
Comment (joeyorlando) [inline engine/apps/schedules/tasks/shift_swaps/slack_followups.py:46]: _nit_ on the query. We already have this same query [here](https://github.com/grafana/oncall/blob/dev/engine/apps/mobile_app/tasks.py#L514). wdyt about moving it as a method on the `QuerySet` (or `Manager`, not sure which one is more appropriate)? Something like, `get_open_requests`.

This way if we need to change the definition of this query, it only needs to be done once.

**Review by joeyorlando — APPROVED** [review]
LGTM! Just one minor nit comment.

File: engine/apps/schedules/tasks/shift_swaps/slack_followups.py (hunk 2 above)
Comment (vadimkerr) [reply engine/apps/schedules/tasks/shift_swaps/slack_followups.py]: Good call, addressed in https://github.com/grafana/oncall/pull/2798/commits/7fe839dd39ceb9706f52ecf1cda17257fbf14555

**Pull request merged**